src/app/ui-library/button.tsx

```diff
@@ -0,0 +1,202 @@
+        `;
+      case 'tertiary':
+        return `
+          color: ${props.theme.colors.white};
+        `;
+      case 'danger':
+        return `
+          background-color: ${props.theme.colors.danger_dark};
+          color: ${props.theme.colors.white};
+        `;
+      default:
+        return '';
+    }
+  }}
+
+  &:focus, &:hover {
+    ${(props) => {
+      if (props.$disabled) return '';
+
+      switch (props.$variant) {
+        case 'primary':
+          return `
+            background-color: ${props.theme.colors.white_200};
+          `;
+        case 'secondary':
+          return `
+            border: 1px solid ${props.theme.colors.white_850};
+            background-color: ${props.theme.colors.elevation6_800};
+          `;
+        case 'tertiary':
+          return `
+            color: ${props.theme.colors.white_200};
+          `;
+        case 'danger':
+          return `
+            background-color: ${props.theme.colors.danger_dark_100};
+          `;
+        default:
+          return '';
+      }
+    }}
+  }
+
+  &:active {
+    ${(props) => {
+      if (props.$disabled) return '';
+
+      switch (props.$variant) {
+        case 'primary':
+          return `
+            background-color: ${props.theme.colors.white_400};
+          `;
+        case 'secondary':
+          return `
+            border: 1px solid ${props.theme.colors.white};
+            background-color: ${props.theme.colors.elevation6_600};
+          `;
+        case 'tertiary':
+          return `
+            color: ${props.theme.colors.white_400};
+          `;
+        case 'danger':
+          return `
+            background-color: ${props.theme.colors.danger_dark_200};
+          `;
+        default:
+          return '';
+      }
+    }}
+  }
+`;
+
+type Props = {
+  title: string;
+  onClick: () => void;
+  icon?: React.ReactNode;
+  disabled?: boolean;
+  loading?: boolean;
+  className?: string;
+  variant?: ButtonVariant;
+};
+
+function Button({
+  title,
+  onClick,
+  icon,
+  className,
+  loading = false,
+  disabled = false,
+  variant = 'primary',
+}: Props) {
+  const theme = useTheme();
+
+  const handleClick = () => {
+    if (!disabled && !loading) {
+      onClick();
+    }
+  };
+
+  const onKeyPressHandler = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    if (e.key === 'Enter') {
+      handleClick();
+    }
+  };
+
+  const spinnerColor =
+    variant === 'primary'
+      ? theme.colors.elevation_n1
+      : variant === 'secondary'
+      ? theme.colors.white_0
+      : variant === 'tertiary'
+      ? theme.colors.white_0
+      : variant === 'danger'
+      ? theme.colors.white_200
+      : theme.colors.elevation_n1;
+
+  return (
+    <ButtonContainer
+      className={className}
+      onClick={handleClick}
+      role="button"
+      tabIndex={0}
+      onKeyDown={onKeyPressHandler}
+      $variant={variant}
+      $disabled={disabled}
+      $loading={loading}
+    >
```

Comment:
we should probably stick to <button> html which comes with a disabled attribute and css selector

Reply:
![image](https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/2c74b4b4-6586-41b7-969d-615887f6691f)

Reply:
Makes a lot of sense 😋 Will fix 👍